app/entrypoint.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/sh
+
+if [[ -z "$CATTLE_URL" ]]; then
+    echo "Error: can't get my CATTLE_URL !" >&2
+    exit 1
+fi
+
+if [[ -z "$CATTLE_ACCESS_KEY" ]]; then
+    echo "Error: can't get my CATTLE_ACCESS_KEY !" >&2
+    exit 1
+fi
+
+if [[ -z "$CATTLE_SECRET_KEY" ]]; then
+    echo "Error: can't get my CATTLE_SECRET_KEY !" >&2
+    exit 1
+fi
+
+function check_writable_directory {
+    local dir="$1"
+    if [[ ! -d "$dir" ]]; then
+        echo "Error: can't access to '$dir' directory !" >&2
+        echo "Check that '$dir' directory is declared has a writable volume." >&2
+        exit 1
+    fi
+    touch $dir/.check_writable 2>/dev/null
+    if [[ $? -ne 0 ]]; then
+        echo "Error: can't write to the '$dir' directory !" >&2
+        echo "Check that '$dir' directory is export as a writable volume." >&2
+        exit 1
+    fi
+    rm -f $dir/.check_writable
+}
+
+function check_dh_group {
+  if [[ ! -f /etc/nginx/dhparam/dhparam.pem ]]; then
+    echo "Creating Diffie-Hellman group (can take several minutes...)"
+    openssl dhparam -out /etc/nginx/dhparam/.dhparam.pem.tmp 2048
```

Comment:
We certainly could. Currently [entrypoint.sh](https://github.com/CausticLab/rgon-proxy/pull/2/files#diff-f4c0396de4314fb0fd536fae34d42438) generates this `dhparam.pem` file in the `check_dh_group` function, so we would need to use parameters in both of these places.

How do you propose parameterization - environment variable?